### ISSUE TITLE
Segmentation fault at game load

### ISSUE THREAD

**Issue by sebaFlame** [Return-To-The-Roots/s25client#290]
Version: 20151009-3c8d2ee
Using dynamic Boost libs (1.55 or 1.56), built in Gentoo.
With this patch:
```
--- CMakeLists.txt      2015-09-28 03:07:04.262222066 +0200
+++ CMakeLists.txt1     2015-09-28 03:08:11.058219909 +0200
@@ -69,7 +69,7 @@

 ################################################################################

-OPTION(USE_STATIC_BOOST "Build with static BOOST libraries instead of dynamic" YES)
+OPTION(USE_STATIC_BOOST "Build with static BOOST libraries instead of dynamic" NO)
 IF((NOT ${Boost_USE_STATIC_LIBS} AND ${USE_STATIC_BOOST}) OR
    (${Boost_USE_STATIC_LIBS} AND NOT ${USE_STATIC_BOOST}))
     set(Boost_USE_STATIC_LIBS ${USE_STATIC_BOOST})
```

GDB output (boost 1.55):
```
Loading "/usr/share/games/S2/GFX/PALETTE/PAL5.BBM": [New Thread 0xb4c4cb40 (LWP 11969)]
[New Thread 0xb556cb40 (LWP 11965)]

Program received signal SIGSEGV, Segmentation fault.
0xb7e4190c in std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::string const&) () from /usr/lib/gcc/i686-pc-linux-gnu/4.9.3/libstdc++.so.6
(gdb) where
#0  0xb7e4190c in std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::string const&) () from /usr/lib/gcc/i686-pc-linux-gnu/4.9.3/libstdc++.so.6
#1  0xb7ebada9 in boost::iostreams::mapped_file_source::open_impl(boost::iostreams::basic_mapped_file_params<boost::iostreams::detail::path> const&) () from /usr/lib/libboost_iostreams.so.1.55.0
#2  0x083e272c in boost::iostreams::mapped_file_source::open<boost::iostreams::detail::path> (
    this=<optimized out>, p=...) at /usr/include/boost/iostreams/device/mapped_file.hpp:406
#3  0x083fa22a in open<std::basic_string<char> > (offset=0, length=4294967295, 
    path="/usr/share/games/S2/GFX/PALETTE/PAL5.BBM", this=0xbfffe6e8)
    at /usr/include/boost/iostreams/device/mapped_file.hpp:416
#4  libsiedler2::loader::LoadBBM (file="/usr/share/games/S2/GFX/PALETTE/PAL5.BBM", items=...)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/libsiedler2/src/LoadBBM.cpp:63
#5  0x083ed0a0 in libsiedler2::Load (file="/usr/share/games/S2/GFX/PALETTE/PAL5.BBM", items=..., 
    palette=palette@entry=0x0)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/libsiedler2/src/libsiedler2.cpp:294
#6  0x081b9f3b in Loader::LoadArchiv (this=this@entry=0xb7641008, 
    pfad="/usr/share/games/S2/GFX/PALETTE/PAL5.BBM", palette=palette@entry=0x0, archiv=...)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/src/Loader.cpp:1101
#7  0x081bb1a0 in Loader::LoadFile (this=this@entry=0xb7641008, 
    filePath="/usr/share/games/S2/GFX/PALETTE/PAL5.BBM", palette=palette@entry=0x0, to=...)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/src/Loader.cpp:1129
#8  0x081be234 in Loader::LoadFile (this=this@entry=0xb7641008, 
    pfad="/usr/share/games/S2/GFX/PALETTE/PAL5.BBM", palette=palette@entry=0x0, 
    load_always=load_always@entry=true)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/src/Loader.cpp:1291
#9  0x081be802 in Loader::LoadFileOrDir (this=0xb7641008, 
    file="/usr/share/games/S2/GFX/PALETTE/PAL5.BBM", file_id=5, load_always=true)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/src/Loader.cpp:157
#10 0x081bf2a1 in Loader::LoadFilesFromArray (this=this@entry=0xb7641008, 
    files_count=files_count@entry=33, files=files@entry=0xbfffed98, load_always=load_always@entry=true)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/src/Loader.cpp:189
#11 0x081bf6f3 in Loader::LoadFilesAtStart (this=0xb7641008)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/src/Loader.cpp:102
#12 0x081e5e76 in GameManager::StartMenu (this=0xa68ac28)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/src/GameManager.cpp:294
#13 0x081e61d4 in GameManager::Start (this=0xa68ac28)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/src/GameManager.cpp:127
#14 0x080790f6 in main (argc=1, argv=0xbffff0b4)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/src/main.cpp:293
(gdb) quit
```

GDB output (boost 1.56):
```
Loading "/usr/share/games/S2/GFX/PALETTE/PAL5.BBM": [New Thread 0xb4c2ab40 (LWP 27156)]
[New Thread 0xb5569b40 (LWP 27155)]

Program received signal SIGSEGV, Segmentation fault.
0xb7e3e90c in std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::string const&) () from /usr/lib/gcc/i686-pc-linux-gnu/4.9.3/libstdc++.so.6
(gdb) where
#0  0xb7e3e90c in std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::string const&) () from /usr/lib/gcc/i686-pc-linux-gnu/4.9.3/libstdc++.so.6
#1  0xb7eb8e36 in boost::iostreams::mapped_file_source::open_impl(boost::iostreams::basic_mapped_file_params<boost::iostreams::detail::path> const&) () from /usr/lib/libboost_iostreams.so.1.56.0
#2  0x083f35bc in boost::iostreams::mapped_file_source::open<boost::iostreams::detail::path> (
    this=<optimized out>, p=...) at /usr/include/boost/iostreams/device/mapped_file.hpp:406
#3  0x0840b16a in open<std::basic_string<char> > (offset=0, length=4294967295, 
    path="/usr/share/games/S2/GFX/PALETTE/PAL5.BBM", this=0xbfffe758)
    at /usr/include/boost/iostreams/device/mapped_file.hpp:416
#4  libsiedler2::loader::LoadBBM (file="/usr/share/games/S2/GFX/PALETTE/PAL5.BBM", items=...)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/libsiedler2/src/LoadBBM.cpp:63
#5  0x083fdda0 in libsiedler2::Load (file="/usr/share/games/S2/GFX/PALETTE/PAL5.BBM", items=..., 
    palette=palette@entry=0x0)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/libsiedler2/src/libsiedler2.cpp:294
#6  0x081ac76f in Loader::LoadArchiv (this=this@entry=0xb763e008, 
    pfad="/usr/share/games/S2/GFX/PALETTE/PAL5.BBM", palette=palette@entry=0x0, archiv=...)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/src/Loader.cpp:1068
#7  0x081ad83f in Loader::LoadFile (this=this@entry=0xb763e008, 
    filePath="/usr/share/games/S2/GFX/PALETTE/PAL5.BBM", palette=palette@entry=0x0, to=...)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/src/Loader.cpp:1096
#8  0x081b26f1 in Loader::LoadFile (this=this@entry=0xb763e008, 
    pfad="/usr/share/games/S2/GFX/PALETTE/PAL5.BBM", palette=0x0, load_always=load_always@entry=true)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/src/Loader.cpp:1268
#9  0x081b2dcf in Loader::LoadFileOrDir (this=0xb763e008, 
    file="/usr/share/games/S2/GFX/PALETTE/PAL5.BBM", file_id=5, load_always=true)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/src/Loader.cpp:156
#10 0x081b3766 in Loader::LoadFilesFromArray (this=this@entry=0xb763e008, files_count=33, 
    files=0xa8ad228, load_always=load_always@entry=true)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/src/Loader.cpp:188
#11 0x081b41bf in Loader::LoadFilesAtStart (this=0xb763e008)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/src/Loader.cpp:101
#12 0x081d8886 in GameManager::StartMenu (this=0xa691d40)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/src/GameManager.cpp:294
#13 0x081d8be4 in GameManager::Start (this=0xa691d40)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/src/GameManager.cpp:127
#14 0x08076966 in main (argc=1, argv=0xbffff0b4)
    at /var/tmp/portage/games-strategy/s25rttr-9999/work/s25rttr-9999/src/main.cpp:293
(gdb) quit
```

**Comment by Flamefire** [comment]
First: No need to patch anything. You can just set this in cmake-gui or ccmake or via cmake -Dfoo=bar

About the crash: This is nothing we can change. Something seems to be wrong with your compiler or boost libs. If you find something that fixes this we will include that.

**Comment by Flamefire** [comment]
My apologies, this bug was caused by an invalid(?) setting in the RTTRs CMake files. This is now fixed in the latest source (see #443)
